langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java

```diff
@@ -56,6 +57,10 @@ public static ToolSpecification toolSpecificationFrom(Method method) {
             if (parameter.isAnnotationPresent(ToolMemoryId.class)) {
                 continue;
             }
+            //if user define method parameter as Optional addOptionalParameter
+            if (parameter.getType().equals(Optional.class)) {
+                builder.addOptionalParameter(parameter.getName(), toJsonSchemaProperties(parameter));
```

Comment:
`toJsonSchemaProperties` will receive `Optional` type, but should probably receive an actual type

Reply:
`dev.langchain4j.agent.tool.DefaultToolExecutor#coerceArgument` also need packe `Optional`?

Also, I have a question: Is `Optional` suitable as a parameter?
[Why should Java 8's Optional not be used in arguments](https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments)

Reply:
I have no strong opinion on this, as always it depends...

```java
@Tool
void method(@P(optional = true) String parameter) {...}
```

or 

```java
@Tool
void method(Optional<String> parameter) {...}
```

what would you choose?

Reply:
I vote for option one~

Reply:
@geoand @jmartisk WDYT?

Reply:
I don't have a super strong opinion either, but I propably slightly lean towards `1`

Reply:
Ok, I guess it does not harm to have both options, but for now tel's go with the first one then